File: 2021/10/15/SmallestRange.java
/**
 * https://leetcode.com/problems/smallest-range-i/
 * @author Andrew Jarombek
 * @since 10/15/2021
 */

public class SmallestRange {
    public int smallestRangeI(int[] nums, int k) {
        int max = nums[0];
        int min = nums[0];

        for (int i = 1; i < nums.length; i++) {
            int val = nums[i];

            if (val < min) {
                min = val;
            }

            if (val > max) {
                max = val;
            }
        }

        return Math.max(0, max - min - (k * 2));
    }
}

File: 2023/05/13/smallestRange.js
/**
 * https://leetcode.com/problems/smallest-range-i/
 * @author Andrew Jarombek
 * @since 5/13/2023
 */

const assert = require('assert');

const smallestRangeI = function(nums, k) {
    let max = -Infinity;
    let min = Infinity;

    for (let num of nums) {
        max = Math.max(max, num);
        min = Math.min(min, num);
    }

    return Math.max(0, max - min - k * 2);
}

let nums = [1, 2, 3, 4, 5];
let k = 2;
let expected = 0;
let result = smallestRangeI(nums, k);
assert.strictEqual(result, expected);

nums = [4, 7, 2, 8, 1];
k = 3;
expected = 1;
result = smallestRangeI(nums, k);
assert.strictEqual(result, expected);
